app/models/compute_resources/foreman/model/ovirt.rb

```diff
@@ -138,6 +174,7 @@ def create_vm(args = {})
       if (image_id = args[:image_id])
         args.merge!({:template => image_id})
       end
+      args[:os] = { :type => determine_os_type(args[:foreman_host]) } if supports_operating_systems?
```

Comment:
I think we mess already a lot with the compute attributes on we're
constantly running into issues regarding to this (I've just hit several
of them last couple of days
http://projects.theforeman.org/issues/14211,http://projects.theforeman.org/issues/14210).
I've also noticed other host method that deals with compute attributes
more than it should
https://github.com/theforeman/foreman/blob/1.11.0-RC2/app/models/concerns/orchestration/compute.rb#L265-L272.

Since the compute attributes are consumed by all the compute
resources, I want to avoid risk of adding something to this generic
hash that might resonate badly with some other compute resource than
ovirt (and every new key in the attribute brings this risk).

Therefore, I've refactored the code a bit to move the responsibility
of preparing the compute attributes from host into the compute
resource. This way, some specific keys can be added to a specific
compute resource without affecting the rest. I agree the host is
a god object right now, but we should rather extract the logic from
host and keep it as data source, then adding more logic to that to
extract the data for particular uses. This way, next time some compute
resource needs some data form host that were not there before, will
just override the `host_compute_attrs` method.

I want to keep this PR focused mainly on the main issue of operating
systems (which is quite painful for many users - the operating system
flag in ovirt influences a lot how the VMs actually behave). But the
steps I've done here are preparation for other refactorings and fixes
that we need to do as well around the compute attributes.